cmd/kube-apiserver/app/server.go

```diff
@@ -94,31 +95,52 @@ cluster's shared state through which all other components interact.`,
 }
 
 // Run runs the specified APIServer.  This should never exit.
-func Run(s *options.ServerRunOptions) error {
-	config, sharedInformers, err := BuildMasterConfig(s)
+func Run(commandOptions *options.ServerRunOptions) error {
+	kubeAPIServerConfig, sharedInformers, err := CreateKubeAPIServerConfig(commandOptions)
+	if err != nil {
+		return err
+	}
+	kubeAPIServer, err := CreateKubeAPIServer(kubeAPIServerConfig, sharedInformers, wait.NeverStop)
 	if err != nil {
 		return err
 	}
 
-	return RunServer(config, sharedInformers, wait.NeverStop)
-}
+	// if we're starting up a hacked up version of this API server for a weird test case,
+	// just start the API server as is because clients don't get built correctly when you do this
+	if len(os.Getenv("KUBE_API_VERSIONS")) > 0 {
+		return kubeAPIServer.GenericAPIServer.PrepareRun().Run(wait.NeverStop)
```

Comment:
Can we wire in a `stopCh` from the beginning? If we need this, we probably also need the stop channel.